tests/conftest.py

```diff
@@ -89,3 +86,112 @@ def _snapshot(**kwargs):
             yield snapshot
 
     return _snapshot
+
+
+if PY2:
+    import marshal
+    from py_compile import MAGIC
+    from py_compile import wr_long
+
+    from _pytest._code.code import ExceptionInfo
+
+    def dump_code_to_file(code, file):
+        file.write(MAGIC)
+        wr_long(file, long(time.time()))  # noqa
+        marshal.dump(code, file)
+        file.flush()
+
+
+else:
+    import importlib
+
+    code_to_pyc = getattr(
+        importlib._bootstrap_external, "_code_to_bytecode" if sys.version_info < (3, 7) else "_code_to_timestamp_pyc"
+    )
+
+    def dump_code_to_file(code, file):
+        file.write(code_to_pyc(code, time.time(), len(code.co_code)))
+        file.flush()
+
+
+def unwind_params(params):
+    if params is None:
+        yield None
+        return
+
+    for _ in product(*(((k, v) for v in params[k]) for k in params)):
+        yield dict(_)
+
+
+def run_function_from_file(item, params=None):
+    file, _, func = item.location
+    marker = item.get_closest_marker("subprocess")
+
+    args = marker.kwargs.get("args", [])
+    env = os.environ.copy()
+    env.update(marker.kwargs.get("env", {}))
+    if params is not None:
+        env.update(params)
+
+    expected_status = marker.kwargs.get("status", 0)
+    expected_out = marker.kwargs.get("out", "").encode("utf-8")
+    expected_err = marker.kwargs.get("err", "").encode("utf-8")
+
+    with open(file) as f:
+        t = ast.parse(f.read())
+        for node in t.body:
+            if isinstance(node, ast.FunctionDef) and node.name == func:
+                t.body = node.body
+                break
+
+    with NamedTemporaryFile(mode="wb", suffix=".pyc") as fp:
+        dump_code_to_file(compile(t, file, "exec"), fp.file)
+
+        start = time.time()
+        out, err, status, _ = call_program(sys.executable, fp.name, *args, env=env)
```

Comment:
we could still use a variant/way to do `ddtrace-run`.

Reply:
maybe a `bin` optional argument? If not passed it defaults to `sys.executable`, otherwise the given binary name is used? E.g.

~~~ python
@pytest.mark.subprocess(bin="ddtrace-run")
def test_blabla():
    ...
~~~

Reply:
you'd still need `sys.executable` since  you have to do `ddtrace-run python script.py` :(

Reply:
Ah right! Maybe prepend then?

Reply:
yeah, or could also probably get away with

`@pytest.mark.subprocess(ddtrace_run=True)` 😆